src/endpoint/s3/s3_rest.js

```diff
@@ -315,7 +315,7 @@ function get_bucket_and_key(req) {
     return {
         bucket,
         // decode and replace hadoop _$folder$ in key
-        key: decodeURIComponent(key.replace('%', '%25')).replace(/_\$folder\$/, '/'),
+        key: decodeURIComponent(encodeURIComponent(key)).replace(/_\$folder\$/, '/'),
```

Comment:
Can you explain why we need to encode and decode?
What does this fix exactly?

Reply:
The original issue was handling the %. 
The previous PR used str replace to replace % with the encoded version.

Reply:
I still don't follow - how does calling `decodeURIComponent(encodeURIComponent(key))` help?
Can you provide a key example and the output?

Reply:
This is what I don't follow in that code:
```
> decodeURIComponent(encodeURIComponent('%'))
'%'
```

Reply:
ok I think I see what happened here. 
key was already decoded in https://github.com/noobaa/noobaa-core/blob/c46671f394647abaac873edea073625853c96537/src/endpoint/s3/s3_rest.js#L299
so this decode(encode(key)) and also replace(/_\$folder\$/, '/') seems to be just duplicates of what was already done in parse_bucket_and_key()...